src/main/java/cn/nukkit/item/ItemBucket.java

```diff
@@ -119,14 +119,17 @@ public boolean onActivate(Level level, Player player, Block block, Block target,
             }
         } else if (targetBlock instanceof BlockLiquid) {
             Item result = Item.get(BUCKET, 0, 1);
-            PlayerBucketEmptyEvent ev;
-            player.getServer().getPluginManager().callEvent(ev = new PlayerBucketEmptyEvent(player, block, face, this, result));
+            PlayerBucketEmptyEvent ev = new PlayerBucketEmptyEvent(player, block, face, this, result);
+
             ev.setCancelled(!block.canBeFlowedInto());
 
-            if (player.getLevel().getName().equals("nether") && this.getDamage() != 10) {
+            if (player.getLevel().getDimension() == Level.DIMENSION_NETHER && this.getDamage() != 10) {
                 ev.setCancelled(true);
             }
 
+            if(!ev.isCancelled())
+                player.getServer().getPluginManager().callEvent(ev);
```

Comment:
Please, make the `callEvent(ev)` happens before `if(!ev.isCancelled())`, so plugins can be notified that the player attempted to empty the bucket and change the cancellation state allowing the placement.